add question backend

JanhaviDadhania opened the pull request:
add practice question backend #34

go to temporary admin_login.php
type 'user' and submit
user will be directed to add_question.php page

checks:

- if any field is empty, it shows message to fill all fields.
- if all details are filled it shows message "successfully added" (wait for 4 seconds and user will be - redirected to main page)
- if question id matches with any other question, it shows message and redirect to main page

updates:

- it updates all testcases and all questions directories
- it updates practice_question table
- it creates new table for added question.

"all testcases" and "all questions" directories should have required permissions to create directory and add files
*update login.php before

JanhaviDadhania committed:
add question backend

add practice question backend

harshraj22 commented on CSE-Hub/admin/add_question.php:
```diff
@@ -0,0 +1,123 @@
+                    <li class="add">Question Id: <input type="text" name="ques_id"></li><br/><br/>
+                    <li class="add">
+                    Question Description: <br/><br/>
+                                      <textarea rows = "5" cols = "50" name = "description">
```
Indentation ??

JanhaviDadhania committed:
added feature to add multiple testcases

harshraj22 commented:
* Check if all fields are filled on front-end (in case of issue, [this](https://stackoverflow.com/questions/17061436/textareas-required-attribute-doesnt-work-even-though-the-value-is-empty) might help ).
* Currently the testcase that is saved contains unexpected tabs at the start of line.
Also adding multiple testcase option might be more helpful.

JanhaviDadhania commented:
- added checks for all in html form (front end)
- indentation and tabs issue solved in previous commit
- added feature to add multiple testcases (at least one ... at most five)

harshraj22 commented:
![Screenshot from 2019-10-08 10-30-18](https://user-images.githubusercontent.com/46635452/66368751-21fc9880-e9b7-11e9-87a4-10d186f900ff.png)

Seems this needs to be fixed as well.

harshraj22 commented on CSE-Hub/admin/add_question.php:
```diff
@@ -78,22 +78,56 @@
+                    <li class="add">Title: <input type="text" name="title" required></li><br/>
+                    <li class="add">Question Id: <input type="text" name="ques_id" required></li><br/><br/>
+                    <li class="add">Date<input type="date" name="date" required></li><br/>
+                    <li class="add">Time Limit in seconds:<input type="number" name="time_limit" required></li><br/>
```
Also, here adding min="0" step="0.5" might be better

DestroyerAlpha commented:
Is the footer issue fixed? It doesn't seem fixed for me.

DestroyerAlpha commented:
![Issue](https://user-images.githubusercontent.com/10846401/66667484-bcd3cc00-ec70-11e9-870c-15b1b530289a.png)
Please check

DestroyerAlpha commented:
Also modify the Error page not found to 404 instead of 4040

JanhaviDadhania commented:
added feature to add more test cases.
Test cases and solutions are stored in the form... ques_id1.txt, sol1.txt and so on...

error page modified.
modified footer styling.

harshraj22 commented on CSE-Hub/admin/update.php:
```diff
@@ -0,0 +1,85 @@
+    $newquery = 'INSERT INTO practice_questions_info (ques_ID, date_created, successful_submissions, total_submissions, admin) VALUES (?, ?, "0", "0", ?)';
+    prepared_query($conn, $newquery, [$ques_id, $date, $username]);
+
+    if(!$newquery)
```
@DestroyerAlpha, @JanhaviDadhania  , Should we check $newquery or $stmt here ? As the result of query execution is stored in $stmt. I think this was missed in the other pr as well. Can you check it out ?

DestroyerAlpha replied:
I think a variable needs to be assigned to the function call.
Need to check though.

JanhaviDadhania committed:
corrected query check

JanhaviDadhania committed:
Merge branch 'develop' into develop

JanhaviDadhania committed:
added sample question

harshraj22 commented on CSE-Hub/practice questions/all questions/q1/q1.txt:
```diff
@@ -1,27 +1,56 @@
+Example Output
+1
+Explanation
+Example case 1: There are six subsequences with length 2: (1,2), (1,3), (1,4), (2,3), (2,4) and (3,4). The minimum sum is 3 and the only subsequence with this sum is (1,2).
```
Things seem repeated here.
